rfcs/0004-pablo-distribution.adoc

```diff
@@ -0,0 +1,480 @@
+
+[plantuml,images/pablo-fee-config,png]
+----
+class FeeConfig {
+    +fee_rate: Permill
+    +owner_fee_rate: Permill,
+    +protocol_fee_rate: Permill,
+}
+----
+
+*Existing code must be modified to use this data structure*.
+
+Given this,
+----
+fee = // calculation depends on the pool type: based on the fee_rate
+owner_fee = fee * owner_fee_rate * (1 - protocol_fee_rate);
+protocol_fee = owner_fee * protocol_fee_rate;
+----
+For all pools launched at the Picasso launch following values would be set for these configs
+----
+owner_fee_rate = 20%
+protocol_fee_rate = 100% // all owner fees goes to composable to be distributed as rewards
+----
+
+==== LP Trading Fee Distribution
+
+LPs trading fees are calculated and kept as part of the pool liquidity in Pablo. When LPs remove liquidity from the pool the trading fees are automatically redeemed according their pool LP ratio, check https://hackmd.io/@HaydenAdams/HJ9jLsfTz#Fee-Structure[reference]. This results in trading fee share being diluted overtime for smaller pools as follows.
+
+[stem]
+++++
+"After " n " trades and " m " liquidity additions,"
+
+"trading fees" = sum a_n
+
+"total liquidity" = sum l_m
+
+"fees and liquidity returned for an LP amount l" = l / (sum l_m) * (sum l_m + sum a_n)
+
+= l + l / (sum l_m) * sum a_n
+
+"trading fees received" = x = l / (sum l_m) * sum a_n
+
+"When pool size " sum l_m " increases the amount of trading fees received " x " reduces for a particular LP position."
+
+"For large pool sizes of " sum l_m " (steady state) this effect is negligible, hence it's a good enough strategy to distribute fees."
+
+"But if required this effect can be negated by increasing the trading fee by a " delta a_n " while at the same time subtracting it from the total fees paid out already to liquidity providers as follows,"
+
+"New trading fee " = x_(adjusted) = l / (sum l_m) * (sum a_n + delta a_n)
+
+"For " r_(th) " liquidity provider,"
+
+delta a = (sum_(r+1) l_m) / (sum l_r) * sum a_n
+
+= x_(adjusted) = l_(r) / (sum_(r+1) l_m + sum l_r) * (sum a_n + (sum_(r+1) l_m) / (sum l_r) * sum a_n )
+
+= l_r / (sum l_r) * sum a_n
+
+"With this adjusted value all later additions to LP shares have been negated when receiving fees for earlier LPs."
+
+
+++++
+
+[#_pblo_staker_trading_fee_distribution]
+==== PBLO Staker Trading Fee Distribution
+
+This is the reward a `PBLO` staker receives from the trading fees of Pablo pools. It is equal to the protocol fee charged on Pablo pools. This can be accomplished by calling the already existing `StakingReward.transfer_reward` interface as follows. According to product there is also a need to convert whatever the fee asset in to PBLO to create a demand/additional value for PBLO.
```

Comment:
i do not think conversion needed. you can make pablo price in LP + lp = total. each each user possition is sum of his pablo with multiplier + lp. if you want more pablo used - raise pablo price. and give away nfts to user to specfici pools kind of. so config is price of palbo vs lp. so you incease rewards share for in pool as one dimensions, and change ration of pablo vs pool token as other dimensions.

Reply:
i more of next
1. you have 1000 USDC + 1000 USDT pool.
2. you say that 20% of fees will go to PBLO stakers and 80% to LP stakers. 
3. it is same as saying that PBLT cost 1/4 LP. 
4. so people are more to stake LP and bring assets.
5. but if say PBLO 80% and 20% LP, than 4x
6. so that people would chace pablo.
7. In code of Pablo pallet, can send rewards to 2 staking pools, using 1/4 ratio.

Reply:
These are configurable params per pool so not relevant to this design